src/giph

```diff
@@ -467,16 +492,37 @@ function giph() {
 
   delete_temporary_directory
   exit 0
+
+}
+function giph() {
+  prepare
+  convert
 }
 
+
 function stop_all_recordings() {
-  for pid in "$(pgrep -f "bash.+giph")"; do
+  for pid in $(pgrep -f "bash.+giph"); do
+    if [ $pid -eq $$ ]; then
```

Comment:
This should not be needed, as long as the INT signal was sent, or am I missing something?

Reply:
In my test if I execute more than one giph programs to record, and use `giph --stop` in order to stop all the previous recording, it will stop only one record. This is because `"$(...)"`  returns a **string** in one single line. So I remove the quotation marks to ensure it is a **array**.

Reply:
I thought this was properly tested when first introduced. This might be one of the times where it just doesn't work like that on your system. But I'm willing to go with your approach, as it does make sense. Maybe some setups handle this differently, we'll see.

Reply:
I'm commenting on the if statement. I get the point you made, but the if statement should be unnecessary.

Reply:
I'm sorry to misunderstand what you are commenting! It's my first time to open a pull request. 

My first thought was to add the if statement to avoid the giph process with `--stop` option having the minimum pid and killing itself first. This may happen when the maximum pid number is consumed out by the system and then the system starts to loop  from 1 to find the first free pid number for use. After learning more about pid, I find the modern 64-bit Linux may have up to $2^{22}$ pid numbers which seems quite enough! So I think it's better to remove the if statement for the tidy of  the code. I will remove it later.